beets/importer.py

```diff
@@ -839,6 +839,19 @@ def reimport_metadata(self, lib):
                         dup_item.id,
                         displayable_path(item.path)
                     )
+                # We exclude certain flexible attributes from the preserving
+                # process since they might have been fetched from MusicBrainz
+                # and been set in beets.autotag.apply_metadata().
+                # discogs_albumid might also have been set but is not a
+                # flexible attribute, thus no exclude is required.
+                if item.get('bandcamp_album_id'):
+                    dup_item.bandcamp_album_id = item.bandcamp_album_id
+                if item.get('spotify_album_id'):
```

Comment:
What is this part doing? preventing overwriting?

Reply:
Just so that I understand, if `spotify_album_id` is already populated, it will not be overwritten?

Reply:
No, you misunderstood what this is doing. I think the commit message explains it but I'll try again in a separate post. First I'll answer to your question:

spotify_album_id as well as the other ID fields AND actually everything else **when imported with beets** will be overwritten. We had discussed this already with Adrian in https://github.com/beetbox/beets/pull/4474. IIRC you were involved in that discussion too :-)

A prevention of something that "is already populated" is difficult to achieve with Beets at its current state. We don't have this feature anywhere. It is hard to realise. I tried it, I failed, but then I listenend to Adrian since he mentioned that we actually don't have that anywhere, thus why not implement this new feature exactly like all the other things that beets imports.

For example: You can't choose whether the `label` field will be not overwritten when you import a record and choose MusicBrainz as the importer source. The option we currently have is: Do not autotag at all (everything imported as-is in the files tags) _or_ *everything* we find in the MusicBrainz release _and_ is supported by beets is overwritten.

Reply:
Ok I hope I did not confuse everyone with that post before. It was complicated but we figured it out together, back then! Decisions were made, we shouldn't wrap those up again. That was the reason why I created a new and hopefully clean PR for this feature :-)

Reply:
So, back to your initial question in this thread: What does this the code-part above actually do:

Short answer: It *enables* this feature *also when remporting*

Long answer (similar to commit message):

- Bandcamp, Spotify, Deezer and Beatport ID's are saved in the library as
  flexible attributes.

- On _reimports_ there is a mechanism (the function above) that makes sure that *existing* values of such flexible attributes (usually they are custom fields applied by the user manually) are not overwritten with empty data! The user would not want that.

- In this case we don't want this behaviour and need to make sure that new
  values are applied. Therefore we check whether such ID's of metadata services
  are present in the reimported items and exclude them in reimport_metadata().

Reply:
Excellent...thanks for the detailed explanation.